docs/reference/api/connectivity/nfc/NDEF.md

```diff
@@ -0,0 +1,106 @@
+using mbed::nfc::ndef::common::Text;
+using mbed::nfc::ndef::common::URI;
+using mbed::nfc::ndef::common::Mime;
+
+using mbed::nfc::ndef::common::SimpleMessageParser;
+
+void parse_ndef_message(const Span<const uint8_t> &buffer) {
+    SimpleMessageParser parser;
+    struct : SimpleMessageParser::Delegate {
+        virtual void on_parsing_started() {
+            printf("parsing started\r\n");
+        }
+
+        virtual void on_text_parsed(const Text &text, const RecordID &) {
+            printf("Text record parsed.\r\n");
+            printf(
+                "\tlanguage: %.*s\r\n",
+                text.get_language_code().size(), text.get_language_code().data()
+            );
+            printf("\ttext: %.*s\r\n",  text.get_text().size(), text.get_text().data());
+        }
+
+        virtual void on_uri_parsed(const URI &uri, const RecordID &) {
+            printf("URI parsed.\r\n");
+            printf("\tid: %d\r\n", uri.get_id());
+            printf("\tvalue: %.*s\r\n",  uri.get_uri_field().size(), uri.get_uri_field().data());
+        }
+
+        virtual void on_mime_parsed(const Mime &mime, const RecordID &) {
+            printf("Mime object parsed.\r\n");
+            printf("\ttype: %.*s\r\n", mime.get_mime_type().size(), mime.get_mime_type().data());
+            printf("\tcontent size: %d\r\n", mime.get_mime_content().size());
+        }
+
+        virtual void on_unknown_record_parsed(const Record &record) {
+            printf("Unknown record parsed.\r\n");
+            printf(
+                "\ttype: %d, type_value: %.*s\r\n",
+                record.type.tnf, record.type.value.size(), record.type.value.data()
+            );
+            printf(
+                "\tpayload size: %d, payload: %.*s\r\n",
+                record.payload.size(), record.payload.size(), record.payload.data()
+            );
+        }
+
+        virtual void on_parsing_terminated() {
+            printf("parsing terminated\r\n");
+        }
+
+        virtual void on_parsing_error(MessageParser::error_t error) {
+            printf("Parsing error: %d\r\n", error);
+        }
+    } delegate;
+
+    parser.set_delegate(&delegate);
+    parser.parse(buffer);
+}
+```
+
+An application can extend capabilities of `SimpleMessageParser` by adding new record parsers (`mbed::nfc::ndef::RecordParser`) at runtime.
+
+[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/FIXME)
+
+#### Construction
+
+The class `mbed::nfc::ndef::MessageBuilder` builds an NDEF message into a user-provided buffer. `URI`, `Text` and `Mime` types can be serialized in the builder with the help of the member function `append_as_record`.
+
+```
+#include "nfc/ndef/MessageBuilder.h"
+
+using mbed::nfc::ndef::MessageBuilder;
+using mbed::nfc::ndef::common::Text;
+using mbed::nfc::ndef::common::URI;
+
+size_t build_ndef_message(const Span<uint8_t> &buffer) {
+    MessageBuilder builder(buffer);
```

Comment:
Same question for this function.
Is this included in Mbed OS or should the developer include this code in their app?

Reply:
Same as above, it is explicitly that `mbed::nfc::ndef::MessageBuilder` is part of the API mbed OS provide. 

It just demonstrate how an application can create an NDEF message.